expo/expo
# [dev-launcher][android] Use the `DevSupportManagerFactory`

# Why

Part of ENG-4210.

# How

Uses `ReactNativeHostHandler` to provide custom `DevSupportManagerFactory` if possible. Otherwise, we fall back to the previous solution.

# Test Plan

- run bare-expo and confirm that the dev support manager is created by the factory (rn 67). 
- create a new project with rn 66 and confirm that the dev support manager is swapped by dev-launcher code
- create a new project with rn 67 and with an old version of the `expo-modules-core` and confirm that the dev support manager is swapped by dev-launcher code
